exporter/graphite/graphite_test.go

```diff
@@ -0,0 +1,359 @@
+		<-time.After(10 * time.Millisecond)
+	}
+
+	str := strings.Trim(string(output), "\n")
+	lines := strings.Split(str, "\n")
+	want := "foo;key1=value1;key2=value2"
+	ok := false
+	for _, line := range lines {
+		if strings.Contains(line, want) {
+			ok = true
+		}
+	}
+
+	if !ok {
+		t.Fatalf("\ngot:\n%s\n\nwant:\n%s\n", output, want)
+	}
+	closeConn = true
+}
+
+func TestMetricsPathOutput(t *testing.T) {
+	exporter, err := NewExporter(Options{Namespace: "opencensus"})
+	if err != nil {
+		t.Fatalf("failed to create graphite exporter: %v", err)
+	}
+
+	closeConn = false
+	output = ""
+	go startServer(exporter)
+
+	view.RegisterExporter(exporter)
+
+	name := "bar"
+
+	var measures mSlice
+	measures.createAndAppend(name, name, "")
+
+	var vc vCreator
+	for _, m := range measures {
+		vc.createAndAppend("foo", m.Description(), nil, m, view.Count())
+	}
+
+	if err := view.Register(vc...); err != nil {
+		t.Fatalf("failed to create views: %v", err)
+	}
+	defer view.Unregister(vc...)
+
+	view.SetReportingPeriod(time.Millisecond)
+
+	for _, m := range measures {
+		stats.Record(context.Background(), m.M(1))
+		<-time.After(10 * time.Millisecond)
+	}
+
+	lines := strings.Split(output, "\n")
+	ok := false
+	for _, line := range lines {
+		if ok {
+			break
+		}
+		for _, sentence := range strings.Split(line, " ") {
+			if sentence == "opencensus.foo" {
+				ok = true
+				break
+			}
+
+			if ok {
+				break
+			}
+		}
+	}
+	if !ok {
+		t.Fatal("path not correct")
+	}
+	closeConn = true
+}
+
+func TestDistributionData(t *testing.T) {
+	exporter, err := NewExporter(Options{Namespace: "opencensus"})
+	if err != nil {
+		t.Fatalf("failed to create graphite exporter: %v", err)
+	}
+	closeConn = false
+	output = ""
+	go startServer(exporter)
+	view.RegisterExporter(exporter)
+	reportPeriod := time.Millisecond
+	view.SetReportingPeriod(reportPeriod)
+
+	m := stats.Float64("tests/bills", "payments by denomination", stats.UnitDimensionless)
+	v := &view.View{
+		Name:        "cash/register",
+		Description: "this is a test",
+		Measure:     m,
+
+		Aggregation: view.Distribution(1, 5, 10, 20, 50, 100, 250),
+	}
+
+	if err := view.Register(v); err != nil {
+		t.Fatalf("Register error: %v", err)
+	}
+	defer view.Unregister(v)
+
+	// Give the reporter ample time to process registration
+	<-time.After(10 * reportPeriod)
+
+	values := []float64{0.25, 245.67, 12, 1.45, 199.9, 7.69, 187.12}
+
+	ctx := context.Background()
+	ms := make([]stats.Measurement, len(values))
+	for _, value := range values {
+		mx := m.M(value)
+		ms = append(ms, mx)
+	}
+	wantLines := []string{
+		`opencensus.cash_register.bucket;le=1 0.25`,
+		`opencensus.cash_register.bucket;le=5 245.92`,
+		`opencensus.cash_register.bucket;le=10 257.91999999999996`,
+		`opencensus.cash_register.bucket;le=20 259.36999999999995`,
+		`opencensus.cash_register.bucket;le=50 459.27`,
+		`opencensus.cash_register.bucket;le=100 466.96`,
+		`opencensus.cash_register.bucket;le=250 654.0799999999999`,
```

Comment:
What about le=+Inf for the last bucket / total?

Reply:
Done